### DISCUSSION TITLE
Community Actor

### DISCUSSION THREAD

**Discussion by outlaw-dame** [assemblee-virtuelle/activitypods#142] [General]
Will ActivityPods support the Community Actor for platforms like Kbin and Lemmy? Lemmy defines it as `ApubCommunity`

**Comment by srosset81** [comment]
Could you tell me what this "Community Actor" is about ? What would be the usage ?
I haven't found mentions of `ApubCommunity` in Lemmy docs. 
I have only found this, but I'm not sure this is what you're referring to:
https://join-lemmy.org/docs/contributors/05-federation.html#community

**Comment by outlaw-dame** [comment]
Sorry , I mislabelled, it is  kinds::actor::GroupType,
  traits::{Actor, Object},
}; https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/objects/community.rs

**Comment by Laurin-W** [comment]
If I understand lemmy here correctly, they seem to be doing mostly what is defined in the [finalised FEP proposal for groups](https://codeberg.org/fediverse/fep/src/branch/main/fep/1b12/fep-1b12.md). This is not implemented right now. That means, that ActivityPods does not support creating Groups and cannot automatically announce (forward) activities to group's followers and so on.

It should however be no problem to follow a Group and send/receive group (community) activities using an ActivityPods account; basically anything that does not involve extra logic on top of following an actor should be okay, since ActivityPods will pass on the activities and supports dynamic json-ld `@context`. Note, I haven't tested interacting with a lemmy community, so I might oversee something.

Since AFAIK lemmy does not implement the client to server AP spec (correct me if I'm wrong), using a lemmy client+API with an ActivityPods account seems to be out of reach, unfortunately. Checking Kbin, some of the docs were offline, when I looked. But it seems to be the same thing.

We might consider implementing the FEP for groups though, so thanks for you question!

**Comment by srosset81** [comment]
On ActivityPods 2.0, it will be possible to handle ActivityPub activities in two ways:
- Natively, directly on the Pod. This is what we do for side-effects such as `Follow` (adding to followers list, etc)
- Through a registered application. On ActivityPods v2, applications will be able to watch inboxes/outboes and act accordingly. So basically any kind of side-effects can be handled this way.

But indeed, we might consider implementing the FEP proposal for group natively in ActivityPods, so that applications don't need to bother about this.
